PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/OrderController.java

```diff
@@ -24,4 +33,10 @@ public void addLinks(final OrderModelDto dto) {
         dto.add(linkTo(methodOn(OrderController.class).delete(dto.getCode())).withRel(DELETE_RELATION_NAME));
         dto.add(linkTo(methodOn(UserController.class).getSingle(dto.getUserCode())).withRel(CUSTOMER_RELATION_NAME));
     }
+
+    @PostMapping("/buy")
+    @PreAuthorize("hasPermission(this.authorizationPrefix, 'post' )")
+    public ResponseEntity saveOrderWithOrderItems(@RequestBody @Valid final List<OrderModelDtoWithOrderItems> orderModelDtoWithItems) {
```

Comment:
@Validated annotation is missing in class. This annotation is not inherited. Pleas verify that.

Reply:
Validation is working without @Validated annotation on that class. Using it does not affect functionality.